lib/from_openstudio/geometry/room.rb

```diff
@@ -64,12 +64,129 @@ def self.properties_from_space(space)
+        hash[:construction_set] = const_set.nameString
+      end
+      unless space.spaceType.empty?
+        space_type = space.spaceType.get
+        hash[:program_type] = space_type.nameString
+      end
+      # TODO: These are loads assigned to the space directly. How should duplicates created in programtype, if any, be handled? 
+      unless space.people.nil?
+        space.people.each do |people|
+          unless people.peopleDefinition.peopleperSpaceFloorArea.empty?
+            hash[:people] = Honeybee::PeopleAbridged.from_load(people)
+          end
+        end
+      end
+      unless space.lights.nil?
+        space.lights.each do |light|
+          unless light.lightsDefinition.wattsperSpaceFloorArea.empty?
+            hash[:lighting] = Honeybee::LightingAbridged.from_load(light)
+          end
+        end
+      end
+      unless space.electricEquipment.nil?
+        space.electricEquipment.each do |electric_eq|
+          electric_eq_def = electric_eq.electricEquipmentDefinition
+          # Only translate if watts per floor area is specified
+          unless electric_eq_def.wattsperSpaceFloorArea.empty?
+            hash[:electric_equipment] = Honeybee::ElectricEquipmentAbridged.from_load(electric_eq)
+          end
+        end
+      end
+      unless space.gasEquipment.nil?
+        space.gasEquipment.each do |gas_eq|
+          hash[:gas_equipment] = Honeybee::GasEquipmentAbridged.from_load(gas_eq)
+        end
+      end
+      #TODO: The translation of other equipment to HB JSON is not working.
+      unless space.otherEquipment.nil?
+        space.otherEquipment.each do |other_eq|
+          hash[:process_loads] = Honeybee::ProcessAbridged.from_load(other_eq)
+        end
+      end
+      unless space.spaceInfiltrationDesignFlowRates.nil?
+        space.spaceInfiltrationDesignFlowRates.each do |infilt|
+          # Only translate if flow per exterior area is specified
+          unless infilt.flowperExteriorSurfaceArea.empty?
+            hash[:infiltration] = Honeybee::InfiltrationAbridged.from_load(infilt)
+          end
+        end
+      end
+      unless space.designSpecificationOutdoorAir.empty?
+        hash[:ventilation] = Honeybee::VentilationAbridged.from_load(space.designSpecificationOutdoorAir.get)
+      end
+      unless space.daylightingControls.empty?
+        space.daylightingControls.each do |daylight_control|
+          hash[:daylighting_control] = Honeybee::DaylightingControl.from_load(daylight_control)
+        end
+      end
+      thermal_zone = space.thermalZone
+      unless thermal_zone.empty?
+        thermal_zone = space.thermalZone.get
+        unless thermal_zone.thermostatSetpointDualSetpoint.empty?
+        # TODO: There isn't a combined setpoint object in OS and the identifier can't be assigned.
+        # For now using the thermal zone name as setpoint identifier.
+          hash[:setpoint] = {}
+          hash[:setpoint][:identifier] = thermal_zone.nameString
```

Comment:
@chriswmackey I have used the thermal zone namestring as an identifier for setpoint

Reply:
Yes that makes sense!